keps/sig-scheduling/2249-pod-affinity-namespace-selector/README.md

```diff
@@ -583,6 +583,25 @@ _This section must be completed when targeting beta graduation to a release._
 * **How can an operator determine if the feature is in use by workloads?**
   The operator can query pods with the NamespaceSelector field set in pod affinity terms.
 
+* **How can someone using this feature know that it is working for their instance?**
```

Comment:
@Huang-Wei @alculquicondor I don't believe we have any status/events/conditions that would indicate that pod-affinity is working, correct? any suggestions how we can address this question?

Reply:
Only the message in the scheduling event if the (anti)affinity is not satisfied.

Reply:
perhaps worth discussing an approach to give more insight on what plugins was evaluated and in case of score how much weight each contributed when a pod successfully schedules.

Reply:
A few options:

- Events: enhance event message with some details. Although, does it make sense to list Filters? I guess you could list only the filters that did a meaningful job (as opposed to passed because the Pod didn't define fields related to the feature). The event message is not too big, so we would have to list the top three scores or something like that.
- Status: We would have to add a new pod condition for Scheduled. The problem is that we would increase one API call for each Pod, which sounds expensive. Recall that today there is one API call for binding and one for the Event.

However, I don't think we would be able to include enough detail for a feature like PodAffinityNamespaceSelector to show up.

Reply:
I am thinking about enhancing the event message, I agree that updating status needs a more thought due to the added api call.

I think we need a new "Success" status that says "NotRelevant", and Filters that returns it would not be included in the message.

> However, I don't think we would be able to include enough detail for a feature like PodAffinityNamespaceSelector to show up.

Right, not directly; but enhancing the event message would still give an indication that the feature is working when a pod using namespace selector passes the filter, we can't do better than that I guess...

Reply:
Including specific reason to event message might involve the least changes. On the other hand, adding more detailed [ErrReasons](https://github.com/kubernetes/kubernetes/blob/2783ddc2274b58e3db0918fddef799e2fe72aae5/pkg/scheduler/framework/plugins/interpodaffinity/filtering.go#L35-L40) on affinity also help, but involves extra changes.

Reply:
while this is all useful, I don't think we can improve the answer to this question much. Let's track it separately?

Reply:
@wojtek-t is this a GA blocker?

Reply:
As long as knowing that would be better than the current state, the main purpose of this question are the "runtime features". 
The metrics shown as SLIs in this case sound like reasonable thing to me, so I'm not going to block on this.

Reply:
sg, this PR is ready for PRR review then.